components/Categories.php

```diff
@@ -74,7 +80,20 @@ public function onRun()
      */
     protected function loadCategories()
     {
-        $categories = BlogCategory::with('posts_count')->getNested();
+        $categoriesQuery = BlogCategory::with('posts_count');
+
+        if ($slug = $this->property('categoryFilter')) {
+            if ((new BlogCategory)->isClassExtendedWith('Winter.Translate.Behaviors.TranslatableModel')) {
+                $categoriesQuery->transWhere('slug', $slug);
+            } else {
+                $categoriesQuery->where('slug', $slug);
+            }
+
+            $categories = $categoriesQuery->first()->getChildren();
```

Comment:
@RomainMazB I'd recommend we continue nesting with `getNested` here, and leave it up to the template whether to display the full nested structure underneath the root category, or just simply show the direct children.